cyclonedx/model/lifecycle.py

```diff
@@ -0,0 +1,240 @@
+# This file is part of CycloneDX Python Library
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) OWASP Foundation. All Rights Reserved.
+
+"""
+    This set of classes represents the lifecycles types in the CycloneDX standard.
+
+.. note::
+    Introduced in CycloneDX v1.5
+
+.. note::
+    See the CycloneDX Schema for lifecycles: https://cyclonedx.org/docs/1.5/#metadata_lifecycles
+"""
+
+from enum import Enum
+from json import loads as json_loads
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Type, Union
+from xml.etree.ElementTree import Element  # nosec B405
+
+import serializable
+from serializable.helpers import BaseHelper
+from sortedcontainers import SortedSet
+
+from .._internal.compare import ComparableTuple as _ComparableTuple
+from ..exception.serialization import CycloneDxDeserializationException
+
+if TYPE_CHECKING:  # pragma: no cover
+    from serializable import ViewType
+
+
+@serializable.serializable_enum
+class Phase(str, Enum):
+    DESIGN = 'design'
+    PREBUILD = 'pre-build'
+    BUILD = 'build'
+    POSTBUILD = 'post-build'
+    OPERATIONS = 'operations'
+    DISCOVERY = 'discovery'
+    DECOMISSION = 'decommission'
+
+
+@serializable.serializable_class
+class PredefinedPhase:
+    """
+    Object that defines pre-defined phases in the product lifecycle.
+
+    .. note::
+        See the CycloneDX Schema definition: https://cyclonedx.org/docs/1.5/#metadata_lifecycles
+    """
+
+    def __init__(self, phase: Phase) -> None:
+        self._phase = phase
+
+    @property
+    def phase(self) -> Phase:
+        return self._phase
+
+    @phase.setter
+    def phase(self, phase: Phase) -> None:
+        self._phase = phase
+
+    def __hash__(self) -> int:
+        return hash(self._phase)
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, PredefinedPhase):
+            return hash(other) == hash(self)
+        return False
+
+    def __lt__(self, other: Any) -> bool:
+        if isinstance(other, PredefinedPhase):
+            return self._phase < other._phase
+        if isinstance(other, CustomPhase):
+            return True  # put PredefinedPhase before any CustomPhase
+        return NotImplemented
+
+    def __repr__(self) -> str:
+        return f'<PredefinedPhase name={self._phase}>'
+
+
+@serializable.serializable_class
+class CustomPhase:
```

Comment:
rename to `CustomLifecycle`?

Reply:
continues in https://github.com/CycloneDX/cyclonedx-python-lib/pull/698#issuecomment-2405665490